WalletLibrary/WalletLibrary/Protocols/Identifier/JWKExportable.swift

```diff
@@ -0,0 +1,16 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+/// A protocol for objects that can export their public key in JWK (JSON Web Key) format.
+/// Types conforming to this protocol provide a way to expose their public key representation.
+protocol JWKExportable
+{
+    /// Exports the public key in `ECPublicJwk` format.
+    ///
+    /// - Returns: An `ECPublicJwk` object representing the public key.
+    /// - Throws: An error if the public key cannot be exported (e.g., due to missing or malformed data).
+    func exportPublicKey() throws -> ECPublicJwk
```

Comment:
Shouldn't this just be `PublicJwk`? We don't require EC in the protocol. Could be RSA or ED25519.

Reply:
yes for sure but this opens a huge can of worms in the code so I'm gunna refactor this to return a better object name in a different PR.